src/cpu/core_dyn_x86/risc_x64.h

```diff
@@ -342,7 +342,8 @@ static BlockReturn gen_runcodeInit(const Bit8u *code) {
 	cache_addb(0xc3);          // ret
 	
 	dyn_mem_execute(cache_addr, cache_bytes);
-	dyn_cache_invalidate(cache_addr, cache_bytes);
+	const size_t cache_flush_bytes = static_cast<size_t>(cache.pos - oldpos);
```

Comment:
More of a question - is there any risk if that subtractions being 0 or negative (`oldpos >= cache.pos`)?

Reply:
The code above always adds a positive number of bytes to the cache. I think we're safe.